oasislmf/computation/generate/losses.py

```diff
@@ -244,6 +244,22 @@ def __check_for_parquet_output(self, analysis_settings, runtypes):
                         )
                     return  # Only need to find a single request
 
+    def __check_summary_group_support(self, analysis_settings, runtypes):
+        """
+        Private method to check the max number of summary groups selected.
+        If that value is greater than 9 then ktools 'summarycalc' will crash.
```

Comment:
It's not that `summarycalc` would crash. For example:
```
$ summarycalc -m -i -10 output_file.bin < input_file.bin
```
would create the files `0` and `0.idx`, for summary group ID 1. Also:
```
$ summarycalc -m -i -20 output_file.bin < input_file.bin
```
would return an error if there was no data with for summary group ID 2.

Perhaps something like this would be more accurate:
```suggestion
        Private method to check the max number of summary groups selected.
        Setting this value greater than 9 will lead to undesirable
        behaviour in ktools 'summarycalc'.
```

Reply:
I think its actually a bit worse, the execution seems to hang after something detects something has gone wrong
```
compute_info : {'allocation_rule': 2, 'max_level': 3, 'max_layer': 1, 'node_len': 72, 'children_len': 102, 'parents_len': 70, 'profile_len': 32, 'loss_len': 111, 'extra_len': 0, 'compute_len': 77, 'start_level': 0, 'items_len': 40, 'output_len': 40, 'stepped': False}
RUNNING: oasislmf.execution.runner.run
/home/sam/repos/models/piwind/runs/losses-20240510110824/run_ktools.sh: line 30: 60453 Killed                  ktools_monitor.sh $$ $LOG_DIR
grep: warning: * at start of expression
grep: warning: * at start of expression

```

Reply:
thats from using: 
```
{                                                                                                                                             
    "version": "3",
    "analysis_tag": "base_example",
    "source_tag": "MDK",
    "model_name_id": "PiWind",
    "model_supplier_id": "OasisLMF",
    "gul_threshold": 0,
    "gul_output": true,
    "model_settings": { 
        "event_set": "p",
        "event_occurrence_id": "lt"
    },
    "gul_summaries": [ 
        { "eltcalc": true, "id": 1 },
        { "eltcalc": true, "id": 2 },
        { "eltcalc": true, "id": 3 },
        { "eltcalc": true, "id": 4 },
        { "eltcalc": true, "id": 5 },
        { "eltcalc": true, "id": 6 },
        { "eltcalc": true, "id": 7 },
        { "eltcalc": true, "id": 8 },
        { "eltcalc": true, "id": 9 },
        { "eltcalc": true, "id": 10 },
        { "eltcalc": true, "id": 11 },
        { "eltcalc": true, "id": 12 } 
    ],
    "il_output": false,
    "il_summaries": [ ],
    "ri_output": false,
    "ri_summaries": [ ] 
}

```

Reply:
Ah, interesting. I just ran `summarycalc` from the command line to see what it would do.

Reply:
I was seeing this in `stderror.err` which is what triggers the kill: 
```
FATAL: openpipe: Summary set ID 1 requested at least twice                                                                                    
FATAL: openpipe: Summary set ID 1 requested at least twice
FATAL: openpipe: Summary set ID 1 requested at least twice
FATAL: openpipe: Summary set ID 1 requested at least twice
FATAL: openpipe: Summary set ID 1 requested at least twice
FATAL: openpipe: Summary set ID 1 requested at least twice
FATAL: openpipe: Summary set ID 1 requested at least twice
  ...
```

Reply:
That makes sense, because it sees `-1` and `-10` as equivalent, i.e. as `-1`. Then I think what you have written in the docstring is fine.